File: web/rest-client/src/test/java/com/tsystems/restclient/GreetingResourceTest.java
package com.tsystems.restclient;

import io.quarkus.test.junit.QuarkusTest;
import io.quarkus.test.junit.mockito.InjectMock;
import org.eclipse.microprofile.rest.client.inject.RestClient;
import org.junit.jupiter.api.Test;
import org.mockito.Mockito;

import static io.restassured.RestAssured.given;
import static org.hamcrest.core.Is.is;

@QuarkusTest
public class GreetingResourceTest {

    @InjectMock
    @RestClient
    GreetingService greetingService;

    @Test
    public void testHelloEndpoint() {
        Mockito.when(greetingService.hello()).thenReturn("hello from mockito");

        given()
                .when().get("/hello")
                .then()
                .statusCode(200)
                .body(is("hello from mockito"));
    }

}

File: alternative_languages/rest-kotlin/src/test/kotlin/com/tsystems/GreetingResourceTest.kt
package com.tsystems

import io.quarkus.test.junit.QuarkusTest
import io.restassured.RestAssured.given
import org.hamcrest.core.IsEqual.equalTo
import org.junit.jupiter.api.Test


@QuarkusTest
class GreetingResourceTest {

    @Test
    fun testHelloEndpoint() {
        given()
                .`when`().get("/greeting")
                .then()
                .statusCode(200)
                .body("message", equalTo("hello"))
    }

}